database/neo4j/neo4j.go

```diff
@@ -70,25 +71,26 @@ func (n *Neo4j) Open(url string) (database.Driver, error) {
 	password, _ := uri.User.Password()
 	authToken := neo4j.BasicAuth(uri.User.Username(), password, "")
 	uri.User = nil
-	uri.Scheme = "bolt"
+	uri.Scheme = "neo4j"
 	msQuery := uri.Query().Get("x-multi-statement")
 
 	// Whether to turn on/off TLS encryption.
 	tlsEncrypted := uri.Query().Get("x-tls-encrypted")
 	multi := false
-	encrypted := false
+
 	if msQuery != "" {
 		multi, err = strconv.ParseBool(uri.Query().Get("x-multi-statement"))
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	if tlsEncrypted != "" {
-		encrypted, err = strconv.ParseBool(tlsEncrypted)
```

Comment:
We need to be backwards compatible, so  we need to handle boolean values being specified. I _think_ the default in v1.8.x was to use `self-signed`.

Docs:
* https://github.com/neo4j/neo4j-go-driver/blob/1.8/neo4j/config.go#L39
* https://neo4j.com/docs/driver-manual/current/client-applications/#driver-connection-uris (See table 1)

Reply:
Yes, have already supported it,
we use Neo4J client version 4
https://github.com/golang-migrate/migrate/blob/b3878eecddc19781b52511d43b48fba02a364c61/database/neo4j/neo4j.go#L78
https://github.com/golang-migrate/migrate/blob/b3878eecddc19781b52511d43b48fba02a364c61/database/neo4j/neo4j.go#L89

Reply:
The current code only supports string values, not boolean values, breaking any existing connection strings.

Reply:
Updated to https://github.com/golang-migrate/migrate/pull/528/commits/4e2ba2fbc4370a21b1366de153af5ca50094cf61
if the `tlsEncrypted` is "true" the uri scheme set to "neo4j+ssc"

Reply:
Use `strconv.ParseBool()` for full backwards compatibility

Reply:
We use Neo4J client version 4, not Neo4J client version 1.X
boolean value was used in Neo4J version 1.X, it's removed in Neo4J version 4.
we can see this in the code below.

This config is removed in Neo4J client version 4
https://github.com/golang-migrate/migrate/pull/528/files#diff-fd7b1dbd62d757f42a6a55c25a7355d22f5290ef52267d24b23bf0864b51df08L105

and this is the approach in Neo4J client version 4
https://github.com/golang-migrate/migrate/pull/528/files#diff-fd7b1dbd62d757f42a6a55c25a7355d22f5290ef52267d24b23bf0864b51df08R88-R93

even though we do not use `strconv.ParseBool()` the function process is the same, 
if the user sets the flag to `true` will set the URI to `neo4j+ssc`.
I think backward compatibility almost impossible because we upgrade the client to version `4`

we can see, `WithInstance` now accept driver from Neo4J client version 4, the user cannot give Neo4J client version 1.x into this function.  
https://github.com/golang-migrate/migrate/pull/528/files#diff-fd7b1dbd62d757f42a6a55c25a7355d22f5290ef52267d24b23bf0864b51df08R49

Reply:
Existing usage of the [CLI](https://github.com/golang-migrate/migrate/tree/master/cmd/migrate) should not break. e.g. users should not need to change the options/args they pass to the CLI after upgrading `migrate`.

Reply:
Yes, CLI will not break,
Neo4J 1.X needs a boolean value, so we parsed it to bool.
On version 4.X doesn't need boolean value, we can use string 'true' or 'false', we don't need to parse it to boolean

Reply:
You're not calling [strconv.parseBool()](https://golang.org/pkg/strconv/#ParseBool), so this is not backwards compatible

Reply:
The reference

https://github.com/golang-migrate/migrate/pull/528/files#diff-fd7b1dbd62d757f42a6a55c25a7355d22f5290ef52267d24b23bf0864b51df08R78,

if I am not mistaken, return from `uri.Query().Get("x-tls-encrypted")` is a string.
but Neo4J version 1.X need boolean value, so we called this func `strconv.parseBool()` converting string `true` or `false` to boolean.

https://github.com/golang-migrate/migrate/pull/528/files#diff-fd7b1dbd62d757f42a6a55c25a7355d22f5290ef52267d24b23bf0864b51df08L105

in Neo4J 4.X this var `config.Encrypted = encrypted` is gone, we doesn't need boolean value anymore,
but we put the `x-tls-encrypted` in

https://github.com/golang-migrate/migrate/pull/528/files#diff-fd7b1dbd62d757f42a6a55c25a7355d22f5290ef52267d24b23bf0864b51df08R89-R93

the `x-tls-encrypted` value can be assigned with `true` or `false`

Reply:
We still need to call `strconv.parseBool()` since it accepts/parses many more boolean values than `true`. From the docs:
> It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.